src/wasm-ops/clip.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "common.h"
+
+extern "C" {
+void clip_f32(void *);
+// Expand for other supported data types for `clip`
+}
+
+// Core implementation of the op
+template <typename T>
+void clip_imp(const T *input, T *output, const int32_t length, const float min,
```

Comment:
No use of templatizing like `binary_imp` as in binary ops there is a lot to templatize on (for example - the bradcasting logic, etc.). Here - there is only a single for loop to templatize on and it becomes cumbersome to deal with attributes in `unary-op` case. So, only templatizing on different `types` of the op and instantiating as necessary.

Reply:
I think it is OK for now. If we are going to have 10+ unary-op implementations it will be the time to consider this problem.

Reply:
I think we can potentially support all unary-ops in WASM. Even then I would find it hard to templatize the core implementation as each has separate attributes. Since there is no native "attribute" object in our WASM layer, it becomes really hard to templatize this.